lib/ecto/schema.ex

```diff
@@ -1369,11 +1369,15 @@ defmodule Ecto.Schema do
     * `:join_where` - A filter for the join table. See "Filtering associations"
       in `has_many/3`
 
-    * `:preload_order` - Sets the default `order_by` of the association.
-      It is used when the association is preloaded.
-      For example, if you set `Post.many_to_many :tags, Tag, join_through: "posts_tags", preload_order: [asc: :foo]`,
-      whenever the `:tags` associations is preloaded, the tags will be order by the `:foo` field.
-      See `Ecto.Query.order_by/3` for more examples.
+    * `:preload_order` - Sets the default `order_by` when preloading the association.
+      It may either be a keyword list/list of fields (see `Ecto.Query.order_by/3`) or
+      a MFA tuple, such as `{Mod, fun, []}`. If specifying a keyword list/list of fields,
+      the ordering applies to the association's table and not the join table. Ordering by
+      the join table can be achieved by specifying a MFA tuple that utilizes
+      `Ecto.Query.dynamic/2`. For example, you may specify `{Mod, fun, []}` where `Mod.fun()`
+      returns `[desc: dynamic([assoc, join], join.updated_at)]`. Please note the ordering of
+      the bindings: the association comes before the join table. If you specify a custom preload
+      query, the bindings for that query come before the join table.
```

Comment:
This is probably too large to keep as a list item. What if we briefly describe the supported inputs and mention a `## Preload order` section, which can go into all of these cases, including examples?